test/new-e2e/tests/remote-config/utils_test.go

```diff
@@ -8,37 +8,91 @@ package remoteconfig
+	// initially all logs are missing
+	copy(missingLogs, expectedLogs)
+	remoteLogsPath := fmt.Sprintf("/var/log/datadog/%s.log", agentName)
+	t.Logf("looking for logs in %s", remoteLogsPath)
 	assert.EventuallyWithTf(t, func(c *assert.CollectT) {
-		output, err := rh.Execute(fmt.Sprintf("cat /var/log/datadog/%s.log", agentName))
-		if assert.NoError(c, err) {
-			assert.Contains(c, output, expectedLogPattern)
+		// read agent logs
+		agentLogs, err := readRemoteFile(rh, remoteLogsPath)
+		if !assert.NoError(c, err) {
+			return
+		}
+		for _, log := range missingLogs {
+			if strings.Contains(agentLogs, log) {
+				t.Logf("found log: %s", log)
+				foundLogs[log] = true
+			}
 		}
-	}, waitFor, tick, "failed to find log with pattern `%s`", expectedLogPattern)
+		// reset missing logs
+		missingLogs = make([]string, 0, len(expectedLogs))
+		for _, log := range expectedLogs {
+			if _, ok := foundLogs[log]; ok {
+				continue
+			}
+			missingLogs = append(missingLogs, log)
+		}
+		assert.Empty(c, missingLogs, "still missing logs")
+		t.Logf("missing logs:\n[%s]", strings.Join(missingLogs, ","))
+	}, waitFor, tick, "failed finding logs in agent")
 }
 
 // mustCurlAgentRcServiceEventually will curl the remote config service's endpoint to get tracer
 // configurations every `tick` until either it is successful (in which case it will return the
 // output of the curl command), or the `waitFor` duration is reached (in which case it will
 // fail the calling test).
-func mustCurlAgentRcServiceEventually(t *testing.T, rh *components.RemoteHost, payload string, waitFor time.Duration, tick time.Duration) string {
+func assertCurlAgentRcServiceContainsEventually(t *testing.T, rh *components.RemoteHost, payload string, expectedKeys []string, waitFor time.Duration, tick time.Duration) string {
 	t.Helper()
 	var output string
+	missingContents := make([]string, len(expectedKeys))
+	copy(missingContents, expectedKeys)
+	foundContents := make(map[string]bool, len(expectedKeys))
 	assert.EventuallyWithTf(t, func(c *assert.CollectT) {
-		curl, err := rh.Execute(fmt.Sprintf("curl -sS localhost:8126/v0.7/config -d @- <<EOF\n%sEOF", payload))
+		curlOutput, err := rh.Execute(fmt.Sprintf("curl -sSL localhost:8126/v0.7/config -d @- <<EOF\n%sEOF", payload))
 		assert.NoError(c, err)
-		output = curl
+		for _, content := range missingContents {
+			if strings.Contains(curlOutput, content) {
+				t.Logf("found content: %s", content)
+				foundContents[content] = true
+			}
+		}
+		// reset missing contents
+		missingContents = make([]string, 0, len(expectedKeys))
+		for _, content := range expectedKeys {
+			if _, ok := foundContents[content]; ok {
+				continue
+			}
+			missingContents = append(missingContents, content)
+		}
+		assert.Empty(c, missingContents, "still missing contents")
 	}, waitFor, tick, "could not curl remote config service")
 	return output
 }
+
+func readRemoteFile(rh *components.RemoteHost, remotePath string) (string, error) {
```

Comment:
Could this be a more generic method (rather than being defined only in the remote-config utils)?

Reply:
Thought about, might work on it in a follow up.